eth2/types/src/beacon_state.rs

```diff
@@ -579,11 +579,23 @@ impl<T: EthSpec> BeaconState<T> {
     /// Gets the state root for some slot.
     ///
     /// Spec v0.6.3
-    pub fn get_state_root(&mut self, slot: Slot) -> Result<&Hash256, Error> {
+    pub fn get_state_root(&self, slot: Slot) -> Result<&Hash256, Error> {
         let i = self.get_latest_state_roots_index(slot)?;
         Ok(&self.latest_state_roots[i])
     }
 
+    /// Gets the oldest (earliest slot) state root.
+    ///
+    /// Spec v0.5.1
+    pub fn get_oldest_state_root(&self) -> Result<&Hash256, Error> {
+        let lookback = std::cmp::min(
```

Comment:
Is this `min` to account for overflow? Would saturating sub be clearer?

Reply:
I'm not sure what I was doing here. Nothing sensible it seems, good find.